### ISSUE TITLE
Error 411 when using Bearer token with detect

### ISSUE THREAD

**Issue by FilipDeVos** [blackducksoftware/blackduck-common#268]
On multiple BD servers we are encountering error 411 when trying to authenticate Detect. When we switch to user/password combo it works fine.

When I call the authentication api from my rest client, everything works. 
```
> POST /api/tokens/authenticate HTTP/1.1
> Host: my.redactedserver.com
> Authorization: token NGE2ZGIsomefunnytokenvaluesmUwYmEwYjU5LTlmYjItNDdlMS1hYWNlLTU2OTQ3OTNjZWJhNA==
> Accept: */*
> Content-Length: 0
```

Response

```
< HTTP/1.1 200 OK
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Transfer-Encoding: chunked
< Content-Type: application/vnd.blackducksoftware.user-4+json;charset=UTF-8
< Expires: 0
< Server: nginx
< X-CSRF-HEADER: X-CSRF-TOKEN
< X-CSRF-PARAM: _csrf
< X-CSRF-TOKEN: ab6710xxxxxxxxxxxxxxxxxd-1f773546423d
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Strict-Transport-Security: max-age=31536000; includeSubDomains
< X-Frame-Options: SAMEORIGIN
< Date: Thu, 19 Sep 2019 18:48:28 GMT
```

When looking through the code I see some tampering with the HTTP Request headers in `ApiTokenBlackDuckHttpClient`, I suspect that adding `headers.put("Content-Length", "0");` will fix the issue

https://github.com/blackducksoftware/blackduck-common/blob/7bc1215c5a45ea63053877c459a189cab6536dfd/src/main/java/com/synopsys/integration/blackduck/rest/ApiTokenBlackDuckHttpClient.java#L77

**Comment by jamesrichard91** [comment]
@FilipDeVos  what version of Black Duck are you using when you are getting these 411 errors? We authenticate using Api tokens all the time without any issues.

**Comment by jamesrichard91** [comment]
What rest client are you using when you test this?

**Comment by jamesrichard91** [comment]
Did you configure the Black Duck server to require a content-length? Or are you using a proxy that requires a content-length?

**Comment by FilipDeVos** [comment]
We are running BLACKDUCK v2019.6.1 and 4.4.1 (in the process of upgrading). We have an nginx reverse proxy in front of the server which most likely is the one rejecting the request. 

I reviewed https://tools.ietf.org/html/rfc7230#section-3.3.2 and although it doesn't say "MUST" i think it would be better to send `Content-Length=0` 
 
>    A user agent SHOULD send a Content-Length in a request message when
>    no Transfer-Encoding is sent and the request method defines a meaning
>    for an enclosed payload body.  For example, a Content-Length header
>    field is normally sent in a POST request even when the value is 0
>    (indicating an empty payload body).  A user agent SHOULD NOT send a
>    Content-Length header field when the request message does not contain
>    a payload body and the method semantics do not anticipate such a
>    body.

**Comment by jamesrichard91** [comment]
This should be resolved in 44.2.5
